src/iface/ethernet.rs

```diff
@@ -241,11 +241,12 @@ impl<'b, 'c, DeviceT> Interface<'b, 'c, DeviceT>
                     Socket::Icmp(ref mut socket) =>
                         socket.dispatch(&caps, |response| {
                             let tx_token = device.transmit().ok_or(Error::Exhausted)?;
-                            match response {
+                            device_result = match response {
                                 (IpRepr::Ipv4(repr), icmp_repr) =>
```

Comment:
Now that I look at it, can any of you explain what this `match` statement does?

Reply:
As far as I understand, since `ICMPv4` and `ICMPv6` are somewhat different (unlike TCP and UDP), this match guarantees we respond to an `ICMPv4` packet with `ICMPv4`. Specifically, we shouldn't return `Packet::Icmpv4` for IPv6 packets.

Reply:
Oh. Can you rename the captured variables to `ipv4_repr` and `icmpv4_repr`? That makes the intent obvious.

Reply:
Done.